### PR TITLE
GW-1326: Ensure Specification Alignment

### PR DESCRIPTION
**Pull request by filip-soldan-cko**
https://checkout.atlassian.net/browse/GW-1326

Compared following objects:
Action 
BillingDescriptor
CaptureRequest
CaptureResponse
CardSourceResponse
Customer
GetPaymentResponse
PaymentDestination
PaymentPending
PaymentProcessed
PaymentRecipient
PaymentRequest
RefundRequest
RefundResponse
Risk
Shipping
ThreeDsEnrollment
VoidRequest
VoidResponse

Shouldn't we move `paymentId `parameters from `IPaymentsClient `to respective objects (like `CaptureRequest`)?

### CODE REVIEW

File: src/CheckoutSdk/Payments/Enums.cs [hunk 1]
```diff
@@ -0,0 +1,34 @@
+﻿namespace Checkout.Payments
+{
+    public enum ActionType
```
Comment (ben-foster-cko) [inline src/CheckoutSdk/Payments/Enums.cs]: Please move into separate files, one for each enumeration.

File: src/CheckoutSdk/Payments/IPaymentsClient.cs [hunk 2]
```diff
@@ -12,6 +12,6 @@ Task<PaymentResponse> RequestAsync<TPaymentSource>(PaymentRequest<TPaymentSource
         Task<CaptureResponse> CaptureAsync(string paymentId, CaptureRequest captureRequest = null, CancellationToken cancellationToken = default(CancellationToken));
         Task<RefundResponse> RefundAsync(string paymentId, RefundRequest refundRequest = null, CancellationToken cancellationToken = default(CancellationToken));
         Task<GetPaymentResponse> GetAsync(string paymentId, CancellationToken cancellationToken = default(CancellationToken));
-        Task<ICollection<ActionProcessed>> GetActionsAsync(string paymentId, CancellationToken cancellationToken = default(CancellationToken));
+        Task<ICollection<Action>> GetActionsAsync(string paymentId, CancellationToken cancellationToken = default(CancellationToken));
```
Comment (ben-foster-cko) [inline src/CheckoutSdk/Payments/IPaymentsClient.cs]: Should be `IEnumerable` not `ICollection`

File: src/CheckoutSdk/Payments/Action.cs [hunk 3]
```diff
@@ -0,0 +1,20 @@
+using Checkout.Common;
+using System;
+using System.Collections.Generic;
+
+namespace Checkout.Payments
+{
+    public class Action : Resource
```
Comment (ben-foster-cko) [inline src/CheckoutSdk/Payments/Action.cs]: Please rename to `PaymentAction`

File: test/CheckoutSdk.Tests/Payments/CardSourcePaymentsTests.cs [hunk 4]
```diff
@@ -299,20 +291,20 @@ public async Task ItCanGetPaymentAction()
             PaymentRequest<CardSource> paymentRequest = TestHelper.CreateCardPaymentRequest();
             PaymentResponse paymentResponse = await _api.Payments.RequestAsync(paymentRequest);
 
-            ICollection<ActionProcessed> actionsResponse = await _api.Payments.GetActionsAsync(paymentResponse.Payment.Id);
+            ICollection<Checkout.Payments.Action> actionsResponse = await _api.Payments.GetActionsAsync(paymentResponse.Payment.Id);
```
Comment (ben-foster-cko) [inline test/CheckoutSdk.Tests/Payments/CardSourcePaymentsTests.cs]: You can remove all these fully qualified types once you rename to `PaymentAction`

File: src/CheckoutSdk/Payments/IPaymentsClient.cs (hunk 2 above)
Comment (filip-soldan-cko) [reply src/CheckoutSdk/Payments/IPaymentsClient.cs]: That way Resharper will complain about possible multiple enumerations.

Will fix in [this PR](https://github.com/CKOTech/checkout-sdk-net/pull/9) to minimize conflicts.

File: src/CheckoutSdk/Payments/Action.cs (hunk 3 above)
Comment (filip-soldan-cko) [reply src/CheckoutSdk/Payments/Action.cs]: Will fix in [this PR](https://github.com/CKOTech/checkout-sdk-net/pull/9) to minimize conflicts.

**Pull request closed without merging**